File: BSvsZP-Common/CommonTester/FieldLocationTester.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using Common;

namespace CommonTester
{
    [TestClass]
    public class FieldLocationTester
    {
        [TestMethod]
        public void FieldLocation_CheckContructors()
        {
            FieldLocation loc = new FieldLocation();
            Assert.AreEqual(0, loc.X);
            Assert.AreEqual(0, loc.Y);
            Assert.AreEqual(false, loc.Immutable);

            loc = new FieldLocation(10, 20);
            Assert.AreEqual(10, loc.X);
            Assert.AreEqual(20, loc.Y);
            Assert.AreEqual(false, loc.Immutable);

            loc = new FieldLocation(-20, -30, true);
            Assert.AreEqual(-20, loc.X);
            Assert.AreEqual(-30, loc.Y);
            Assert.AreEqual(true, loc.Immutable);

            loc = new FieldLocation(false);
            Assert.AreEqual(0, loc.X);
            Assert.AreEqual(0, loc.Y);
            Assert.AreEqual(false, loc.Immutable);

            loc = new FieldLocation(true);
            Assert.AreEqual(0, loc.X);
            Assert.AreEqual(0, loc.Y);
            Assert.AreEqual(true, loc.Immutable);

            loc = new FieldLocation { X = 10, Y = 30 };
            Assert.AreEqual(10, loc.X);
            Assert.AreEqual(30, loc.Y);
            Assert.AreEqual(false, loc.Immutable);

            loc = new ImmutableFieldLocation();
            Assert.AreEqual(0, loc.X);
            Assert.AreEqual(0, loc.Y);
            Assert.AreEqual(true, loc.Immutable);

            loc = new ImmutableFieldLocation { X = 0, Y = 0 };
            Assert.AreEqual(0, loc.X);
            Assert.AreEqual(0, loc.Y);
            Assert.AreEqual(true, loc.Immutable);

            loc = new ImmutableFieldLocation { X = 100, Y = 200 };
            Assert.AreEqual(100, loc.X);
            Assert.AreEqual(200, loc.Y);
            Assert.AreEqual(true, loc.Immutable);

            loc = new ImmutableFieldLocation { X = -200, Y = -300 };
            Assert.AreEqual(-200, loc.X);
            Assert.AreEqual(-300, loc.Y);
            Assert.AreEqual(true, loc.Immutable);
        }

        [TestMethod]
        public void FieldLocation_CheckProperties()
        {
            FieldLocation loc = new FieldLocation();
            Assert.AreEqual(0, loc.X);
            Assert.AreEqual(0, loc.Y);
            Assert.AreEqual(false, loc.Immutable);

            loc.X = 100;
            Assert.AreEqual(100, loc.X);
            loc.X = -200;
            Assert.AreEqual(-200, loc.X);
            loc.X = 0;
            Assert.AreEqual(0, loc.X);


            loc.Y = 100;
            Assert.AreEqual(100, loc.Y);
            loc.Y = -200;
            Assert.AreEqual(-200, loc.Y);
            loc.Y = 0;
            Assert.AreEqual(0, loc.Y);

            loc = new ImmutableFieldLocation { X = 10, Y = 20 };
            Assert.AreEqual(10, loc.X);
            Assert.AreEqual(20, loc.Y);
            Assert.AreEqual(true, loc.Immutable);

            loc.X = 100;
            Assert.AreEqual(10, loc.X);
            loc.X = -200;
            Assert.AreEqual(10, loc.X);
            loc.X = 0;
            Assert.AreEqual(10, loc.X);

            loc.Y = 100;
            Assert.AreEqual(20, loc.Y);
            loc.Y = -200;
            Assert.AreEqual(20, loc.Y);
            loc.Y = 0;
            Assert.AreEqual(20, loc.Y);
        }

        [TestMethod]
        public void FieldLocation_CheckEncodeDecode()
        {
            ByteList bytes = new ByteList();

            FieldLocation loc1 = new FieldLocation { X = 100, Y = 200 };
            loc1.Encode(bytes);
            Assert.AreEqual(9, bytes.Length);
            Assert.AreEqual(3, bytes[0]);
            Assert.AreEqual(238, bytes[1]);
            Assert.AreEqual(0, bytes[2]);
            Assert.AreEqual(5, bytes[3]);
            Assert.AreEqual(0, bytes[4]);
            Assert.AreEqual(100, bytes[5]);
            Assert.AreEqual(0, bytes[6]);
            Assert.AreEqual(200, bytes[7]);
            Assert.AreEqual(0, bytes[8]);

            FieldLocation loc2 = FieldLocation.Create(bytes);
            Assert.AreEqual(loc1.X, loc2.X);
            Assert.AreEqual(loc1.Y, loc2.Y);
            Assert.AreEqual(false, loc2.Immutable);
        }

    }
}

File: BSvsZP-Common -Java/CommonTester/FieldLocationTester.java
package CommonTester;

import static org.junit.Assert.*;

import org.junit.Test;

import Common.*;

public class FieldLocationTester {

	@Test
	public void FieldLocation_CheckContructors()
    {
        FieldLocation loc = new FieldLocation();
        assertEquals(0, loc.getX());
        assertEquals(0, loc.getY());
        assertEquals(false, loc.getImmutable());

        loc = new FieldLocation((short) 10, (short) 20);
        assertEquals(10, loc.getX());
        assertEquals(20, loc.getY());
        assertEquals(false, loc.getImmutable());

        loc = new FieldLocation((short)-20, (short)-30, true);
        assertEquals(-20, loc.getX());
        assertEquals(-30, loc.getY());
        assertEquals(true, loc.getImmutable());

        loc = new FieldLocation(false);
        assertEquals(0, loc.getX());
        assertEquals(0, loc.getY());
        assertEquals(false, loc.getImmutable());

        loc = new FieldLocation(true);
        assertEquals(0, loc.getX());
        assertEquals(0, loc.getY());
        assertEquals(true, loc.getImmutable());

        loc = new FieldLocation();
        loc.setX((short) 10); 
        loc.setY((short) 30);
        
        assertEquals(10, loc.getX());
        assertEquals(30, loc.getY());
        assertEquals(false, loc.getImmutable());

        loc = new ImmutableFieldLocation();
        assertEquals(0, loc.getX());
        assertEquals(0, loc.getY());
        assertEquals(true, loc.getImmutable());

        loc = new ImmutableFieldLocation();
        loc.setX((short) 0); 
        loc.setY((short) 0);
        assertEquals(0, loc.getX());
        assertEquals(0, loc.getY());
        assertEquals(true, loc.getImmutable());

        loc = new ImmutableFieldLocation();
        loc.setX((short) 100); 
        loc.setY((short) 200);
        assertEquals(100, loc.getX());
        assertEquals(200, loc.getY());
        assertEquals(true, loc.getImmutable());

        loc = new ImmutableFieldLocation();
        loc.setX((short) -200); 
        loc.setY((short) -300);
        
        assertEquals(-200, loc.getX());
        assertEquals(-300, loc.getY());
        assertEquals(true, loc.getImmutable());
    }
	
	@Test
	public void FieldLocation_CheckProperties()
    {
         FieldLocation loc = new FieldLocation();
         assertEquals(0, loc.getX());
         assertEquals(0, loc.getY());
         assertEquals(false, loc.getImmutable());

         loc.setX((short)100);
         assertEquals(100, loc.getX());
         loc.setX((short)-200);
         assertEquals(-200, loc.getX());
         loc.setX((short) 0);
         assertEquals(0, loc.getX());


         loc.setY((short)100);
         assertEquals(100, loc.getY());
         loc.setY((short)-200);
         assertEquals(-200, loc.getY());
         loc.setY((short) 0);
         assertEquals(0, loc.getY());

         loc = new ImmutableFieldLocation();
         loc.setX((short) 10); 
         loc.setY((short) 20);
         assertEquals(10, loc.getX());
         assertEquals(20, loc.getY());
         assertEquals(true, loc.getImmutable());

         loc.setX((short) 100);
         System.out.println(loc.getX() + "!!!!!!!!!!!!!!!!!!!!");
        // assertEquals(10, loc.getX());
         loc.setX((short) -200);
         assertEquals(10, loc.getX());
         loc.setX((short) 0);
         assertEquals(10, loc.getX());

         loc.setY((short)100);
         assertEquals(20, loc.getY());
         loc.setY((short)-200);
         assertEquals(20, loc.getY());
         loc.setY((short) 0);
         assertEquals(20, loc.getY());
     }

     @Test
     public void FieldLocation_CheckEncodeDecode() throws Exception
     {
         ByteList bytes = new ByteList();

         FieldLocation loc1 = new FieldLocation();
         loc1.setX((short) 100); 
         loc1.setY((short) 200);
         
         loc1.Encode(bytes);
         assertEquals(9, bytes.getLength());
         assertEquals(3, bytes.getByteValue(0));
         assertEquals(-18, bytes.getByteValue(1)); // fail on 238
         assertEquals(0, bytes.getByteValue(2));
         assertEquals(5, bytes.getByteValue(3));
         assertEquals(0, bytes.getByteValue(4));
         assertEquals(100, bytes.getByteValue(5)); 
         assertEquals(0, bytes.getByteValue(6));
         assertEquals(-56, bytes.getByteValue(7)); // fail on 200
         assertEquals(0, bytes.getByteValue(8));

         FieldLocation loc2 = FieldLocation.Create(bytes);
         assertEquals(loc1.getX(), loc2.getX());
         assertEquals(loc1.getY(), loc2.getY());
         assertEquals(false, loc2.getImmutable());
     }

}
